### ISSUE TITLE
GetAuthorizeUrl throws EpiOAuthException

### ISSUE THREAD

**Issue by Toumal** [jmathai/twitter-async#170]
Note, this is not identical to the other issue about GetAuthorizeUrl.

My problem is that we're getting an empty response from the request to https://api.twitter.com/oauth/request_token - or at least both messageText and code are returned empty.

I've changed the URL to https, tried API version 1 and 1.1, I tried with oauth_callback in the request and without. I tried checking curl for errors and nothing. Curl is working and supports https (we use it for amazon SES as well...) and besides, this has been working before without change. I've also tried the recent changes in other similar topics and none worked. I also checked whether we're rate-limited but this seems not to be the case. 

What happens is that the curl request returns empty and with a code of 0, which causes EpiOAuth.php to raise an EpiOAuthException in class EpiOAuthResponse function __get, as the code is expected to equal 200 there.

Does anyone have any pointers as to what could be the cause? Do note that all this has been working perfectly in the past. We were running the latest twitter-async code with API 1 without problems.

I'd really appeciate any ideas and hints!

**Comment by huangho** [comment]
Wow thax a lot!!!!!

**Comment by Rohit291520** [comment]
Thanks..
I spent one day debugging this. Finally got solution..
Thanks very much :+1:

**Comment by rinogo** [comment]
This "bug" (?) persists to this day, except the code has changed slightly.  The solution is similar: remove/comment out the following two lines in EpiOAuth.php:

```
    if($this->serverAddr !== false)
      curl_setopt($ch, CURLOPT_INTERFACE, $this->serverAddr);
```

Also, for the sake of searchability, here's the output I was getting when I enabled verbose output in cURL:

```
* About to connect() to api.twitter.com port 443 (#0)
*   Trying 104.244.42.2... * Name '192.168.56.2' family 2 resolved to '192.168.56.2' family 2
* Local port: 0
* Timeout
*   Trying 104.244.42.66... * Name '192.168.56.2' family 2 resolved to '192.168.56.2' family 2
* Local port: 0
* Timeout
*   Trying 104.244.42.130... * Name '192.168.56.2' family 2 resolved to '192.168.56.2' family 2
* Local port: 0
* Timeout
*   Trying 104.244.42.194... * Name '192.168.56.2' family 2 resolved to '192.168.56.2' family 2
* Local port: 0
* Timeout
* connect() timed out!
* Closing connection #0

```
